File: scripts/simControl.py
#!/usr/bin/env python
#coding=utf-8
import rospy
from wrt.msg import simControl

rospy.init_node('simControl_node')

simControl_pub = rospy.Publisher('simControl', simControl, queue_size = 1)

# 频率待定
rate = rospy.Rate(2)

while not rospy.is_shutdown():
    simControl_msg = simControl()
    # 初始化数据...


    simControl_pub.publish(simControl_msg)
    rate.sleep()

File: src/node/simControl/simControl.cc
#include <ros/ros.h>
#include <string.h>

#include "wrt/simControl.h"



int main(int argc, char** argv) {
  ros::init(argc, argv, "simControlNode");  // 节点名称
  ros::NodeHandle n;
  ros::Publisher simControlPub =
      n.advertise<wrt::simControl>("simControlTopic", 1000);
  ros::Rate loop_rate(10);

  while (ros::ok()) {
    wrt::simControl simControlMsg;
    //填充数据

    simControlPub.publish(simControlMsg);

    ros::spinOnce();

    loop_rate.sleep();
  }

  return 0;
}
